File: OpenDataAccess/OpenDataAccess.Data/OracleDataBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OpenDataAccess.Data
{
    internal sealed class OracleDataBase : DataBase, IOracleDataBase
    {
        public OracleDataBase()
        {
            this.ClientType = ServerClient.SqlClient;
        }
    }
}

File: Water/Api/src/main/OpenDataAccess/Data/OracleDataBase.java
package OpenDataAccess.Data;

public class OracleDataBase extends DataBase implements IOracleDataBase {

    public OracleDataBase(String connectionString, String user, String password) {
        this.SetClientType(ServerClient.OracleClient);
        this.SetConnectionString(connectionString);
        this.SetUser(user);
        this.SetPassword(password);
    }
}
